src/components/UI/Modals/DeleteModal/index.tsx

```diff
@@ -38,37 +41,51 @@ const deleteModalStyles: Style = {
   },
 };
 
-export const DeleteModal = ({ visible, onCancel }: IProps) => (
-  <GenericModal
-    visible={visible}
-    onCancel={onCancel}
-    modalStyles={deleteModalStyles.modal}
-  >
-    <Grid container sx={deleteModalStyles.innerContainer}>
-      <Grid item sx={deleteModalStyles.heading} xs={12}>
-        <Typography variant="h4" component="h5" color="white" fontWeight={500}>
-          Alert
-        </Typography>
-        <IconButton onClick={onCancel} sx={deleteModalStyles.icon}>
-          <CloseIcon sx={globalStyles.whiteColor} />
-        </IconButton>
-      </Grid>
-      <Grid item xs={12} mt={1}>
-        <Typography>Are you sure you want to delete?</Typography>
-      </Grid>
-      <Grid item xs={12} mt={3}>
-        <Button
-          variant="contained"
-          color="error"
-          onClick={onCancel}
-          sx={{ mr: 3 }}
-        >
-          Delete
-        </Button>
-        <Button variant="contained" onClick={onCancel}>
-          Cancel
-        </Button>
+export const DeleteModal = ({ visible, onCancel, product }: IProps) => {
+  const { deleteProduct } = useProduct();
+
+  const handleDelete = () => {
+    deleteProduct(product);
+    onCancel();
```

Comment:
Move to the handler into useProduct the onCancel + useCallback

Reply:
dont understand

Reply:
i chose to separate